integrated_channels/canvas/views.py

```diff
@@ -49,29 +51,43 @@ class CanvasCompleteOAuthView(generics.ListAPIView):
             a 404 error is returned.
 
     """
+    renderer_classes = [JSONRenderer, ]
+
+    def render_page(self, request, status_code):
+        """
+        Return a success or failure page based on Canvas OAuth response
+        """
+        success_template = 'enterprise/admin/oauth_authorization_successful.html'
+        error_template = 'enterprise/admin/oauth_authorization_failed.html'
+        template = success_template if status_code == 200 else error_template
+
+        return render(request, template, context={})
+
     def get(self, request, *args, **kwargs):
         app_config = apps.get_app_config('canvas')
         canvas_oauth_token_path = app_config.oauth_token_auth_path
 
         # Check if Canvas encountered an error when generating the oauth code.
         canvas_request_error = request.GET.get('error')
         if canvas_request_error:
-            raise APIException(
+            LOGGER.exception(
                 'Canvas OAuth API encountered an error when generating client code- error: {} description: {}'.format(
                     canvas_request_error,
-                    request.GET.get('error_description')
-                )
+                    request.GET.get('error_description'))
             )
+            return self.render_page(request, 'error')
 
         # Retrieve the newly generated code and state (Enterprise user's ID)
         client_code = request.GET.get('code')
         state_uuid = request.GET.get('state')
 
         if not state_uuid:
-            raise ParseError("Canvas Configuration uuid required to integrate with Canvas.")
+            LOGGER.exception("Canvas Configuration uuid required to integrate with Canvas.")
+            return self.render_page(request, 'error')
 
         if not client_code:
-            raise ParseError("Client code required to integrate with Canvas.")
+            LOGGER.exception("Client code required to integrate with Canvas.")
+            return self.render_page(request, 'error')
```

Comment:
are these second params after request meant to be integer status codes or string? looks like in the function it looks for status codes? 

https://github.com/openedx/edx-enterprise/pull/1505/files#diff-733346bb5d0a0dc4dda06b7f2f8be7ac3a7da7019c919ee777cbcf8c57f948a5R80